ocs_ci/ocs/ocs_upgrade.py

```diff
@@ -0,0 +1,376 @@
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_CEPHFSPLUGIN_LABEL,
+        count=number_of_worker_nodes,
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_CEPHFSPLUGIN_PROVISIONER_LABEL,
+        count=2
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_RBDPLUGIN_LABEL,
+        count=number_of_worker_nodes,
+    )
+    verify_pods_upgraded(
+        old_images, selector=constants.CSI_RBDPLUGIN_PROVISIONER_LABEL,
+        count=2
+    )
+    verify_pods_upgraded(old_images, selector=constants.MON_APP_LABEL, count=3)
+    verify_pods_upgraded(old_images, selector=constants.MGR_APP_LABEL)
+    verify_pods_upgraded(
+        old_images, selector=constants.OSD_APP_LABEL, count=osd_count,
+        timeout=1800,
+    )
+    verify_pods_upgraded(old_images, selector=constants.MDS_APP_LABEL, count=2)
+    if config.ENV_DATA.get('platform') == constants.VSPHERE_PLATFORM:
+        verify_pods_upgraded(
+            old_images, selector=constants.RGW_APP_LABEL, count=1
+        )
+
+
+class OCSUpgrade(object):
+    """
+    OCS Upgrade helper class
+
+    """
+
+    namespace = config.ENV_DATA['cluster_namespace']
+    version_before_upgrade = config.ENV_DATA.get("ocs_version")
+    ocs_registry_image = config.UPGRADE.get('upgrade_ocs_registry_image')
+    parsed_version_before_upgrade = parse_version(version_before_upgrade)
+    upgrade_in_current_source = config.UPGRADE.get(
+        'upgrade_in_current_source', False
+    )
+    upgrade_version = config.UPGRADE.get(
+        "upgrade_ocs_version", version_before_upgrade
+    )
+    if ocs_registry_image:
+        upgrade_version = get_ocs_version_from_image(
+            ocs_registry_image
+        )
+    parsed_upgrade_version = parse_version(upgrade_version)
+
+    def load_version_config_file(self, upgrade_version):
+        """
+        Loads config file to the ocs-ci config with upgrade version
+
+        Args:
+            upgrade_version: (str): version to be upgraded
+
+        """
+
+        version_change = self.verify_not_downgrade(upgrade_version)
+        if version_change:
+            version_config_file = os.path.join(
+                constants.CONF_DIR, 'ocs_version', f'ocs-{upgrade_version}.yaml'
+            )
+            load_config_file(version_config_file)
+
+    def verify_not_downgrade(self, upgrade_version):
+        """
+        Checks if upgrade version is newer than current version
+
+        Args:
+            upgrade_version: (str): version to be upgraded
+
+        Returns:
+            bool: True if upgrade version is newer than current version
+
+        """
+        assert self.parsed_upgrade_version >= self.parsed_version_before_upgrade, (
+            f"Version you would like to upgrade to: {upgrade_version} "
+            f"is not higher or equal to the version you currently running: "
+            f"{self.version_before_upgrade}"
+        )
+
+        return True
```

Comment:
the function doesn't return False. Hence, `return` is not needed

Reply:
fixed